src/libraries/Common/src/System/Net/ArrayBuffer.cs

```diff
@@ -57,11 +57,13 @@ public void Dispose()
 
         public int ActiveLength => _availableStart - _activeStart;
         public Span<byte> ActiveSpan => new Span<byte>(_bytes, _activeStart, _availableStart - _activeStart);
-        public ReadOnlySpan<byte> ActiveReadOnlySpan => new ReadOnlySpan<byte>(_bytes, _activeStart, _availableStart - _activeStart);
-        public int AvailableLength => _bytes.Length - _availableStart;
-        public Span<byte> AvailableSpan => new Span<byte>(_bytes, _availableStart, AvailableLength);
         public Memory<byte> ActiveMemory => new Memory<byte>(_bytes, _activeStart, _availableStart - _activeStart);
-        public Memory<byte> AvailableMemory => new Memory<byte>(_bytes, _availableStart, _bytes.Length - _availableStart);
+
+        public int AvailableLength => _bytes.Length - _availableStart;
+        public Span<byte> AvailableSpan => _bytes.AsSpan(_availableStart);
+        public Memory<byte> AvailableMemory => _bytes.AsMemory(_availableStart);
```

Comment:
Any perf difference using `.As` vs `new`? I think I've talked about this before with @GrabYourPitchforks, something about eliding null ref checks?

Reply:
For the overload that just takes an offset, it's ever so slightly cheaper to use As.  If you're passing both offset and length, they're the same, and we just save the JIT a bit of inlining work by using the ctor directly.  It's all minutia, though.